px4_ros2_cpp/include/px4_ros2/navigation/experimental/local_position_measurement_interface.hpp

```diff
@@ -0,0 +1,145 @@
+ * @see LocalPositionMeasurementInterface::update
+ */
+struct LocalPositionMeasurement
+{
+  /** @brief Timestamp of the sample. */
+  rclcpp::Time timestamp_sample {};
+
+  /** @brief Position in the xy-plane. */
+  std::optional<Eigen::Vector2f> position_xy {std::nullopt};
+  /** @brief Variance of position error in the xy-plane. */
+  std::optional<Eigen::Vector2f> position_xy_variance {std::nullopt};
+  /** @brief Position in the z-axis. */
+  std::optional<float> position_z {std::nullopt};
+  /** @brief Variance of position error in the z-axis. */
+  std::optional<float> position_z_variance {std::nullopt};
+
+  /** @brief Velocity in the xy-plane. */
+  std::optional<Eigen::Vector2f> velocity_xy {std::nullopt};
+  /** @brief Variance of velocity error in the xy-plane. */
+  std::optional<Eigen::Vector2f> velocity_xy_variance {std::nullopt};
+  /** @brief Velocity in the z-axis. */
+  std::optional<float> velocity_z {std::nullopt};
+  /** @brief Variance of velocity error in the z-axis. */
+  std::optional<float> velocity_z_variance {std::nullopt};
+
+  /** @brief Attitude quaternion [w, x, y, z], Hamiltonian convention. */
+  std::optional<Eigen::Quaternionf> attitude_quaternion {std::nullopt};
+  /** @brief Variance of attitude error in body frame. */
+  std::optional<Eigen::Vector3f> attitude_variance {std::nullopt};
+};
+
+class LocalPositionMeasurementInterface : public NavigationInterfaceBase
+{
+public:
+  explicit LocalPositionMeasurementInterface(
+    rclcpp::Node & node, PoseFrame pose_frame,
+    VelocityFrame velocity_frame);
+  ~LocalPositionMeasurementInterface() override = default;
+
+  /**
+   * @brief Publishes a local position measurement to the FMU.
+   * The following are checked about the given local position measurement:
+   * 1. The sample timestamp is defined.
+   * 2. Values do not have a NAN.
+   * 3. If an measurement value is provided, its associated variance value is well defined.
+   * 4. If an measurement value is provided, its associated reference frame is not unknown.
+   * @param local_position_measurement The local position measurement to publish.
+   */
+  void update(const LocalPositionMeasurement & local_position_measurement) const;
+
+private:
+  /**
+   * @brief Check that at least one measurement value is defined.
+   */
+  bool isMeasurementNonEmpty(const LocalPositionMeasurement & measurement) const;
+
+  /**
+   * @brief Check that if an measurement value is defined, its variance is also defined and strictly greater than zero.
+   */
+  bool isVarianceValid(const LocalPositionMeasurement & measurement) const;
+
+  /**
+   * @brief Check that if an measurement value is defined, its associated frame is not *FRAME_UNKNOWN.
+   */
+  bool isFrameValid(const LocalPositionMeasurement & measurement) const;
+
+  /**
+   * @brief Check that if an measurement value is defined, none of its fields are NAN.
```

Comment:
```suggestion
   * @brief Check that if a measurement value is defined, none of its fields are NAN.
```

Being a bit pedantic, but find-and-replace unfortunately leaves behind the old articles for every instance of "measurement"